knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -13,598 +13,618 @@
-        user_mail: The user (e-mail) which the ontology should be created with (requesting user)
-        password: The password for the user (requesting user)
-        verbose: Prints more information if set to True
-        dump: Dumps test files (JSON) for DSP API requests if set to True
+        input_file: the path to the JSON file from which the project and its parts should be created
+        server: the URL of the DSP server on which the project should be created
+        user_mail: a username (e-mail) who has the permission to create a project
+        password: the user's password
+        verbose: prints more information if set to True
+        dump: dumps test files (JSON) for DSP API requests if set to True
 
     Returns:
-        None
+        True if everything went smoothly, False if a warning or error occurred
     """
 
     knora_api_prefix = "knora-api:"
+    overall_success = True
+    success = True
 
-    # read the ontology from the input file
+    # create project
+    ################
     with open(input_file) as f:
-        onto_json_str = f.read()
-
-    data_model = json.loads(onto_json_str)
+        project_json_str = f.read()
+    project_definition = json.loads(project_json_str)
+    print(f"Create project '{project_definition['project']['shortname']}' "
+          f"({project_definition['project']['shortcode']})...")
 
-    # expand all lists referenced in the list section of the data model and add them to the ontology
-    data_model["project"]["lists"] = expand_lists_from_excel(data_model)
+    # expand all lists referenced in the "lists" section of the project, and add them to the project
+    new_lists, success = expand_lists_from_excel(project_definition["project"].get("lists", []))
+    if new_lists:
+        project_definition["project"]["lists"] = new_lists
+    if not success:
+        overall_success = False
 
-    # validate the ontology
-    if not validate_ontology(data_model):
-        exit(1)
+    if validate_project(project_definition, expand_lists=False):
+        print('\tJSON project file is syntactically correct and passed validation.')
 
-    # make the connection to the server
     con = login(server=server, user=user_mail, password=password)
-
     if dump:
         con.start_logging()
 
     # read the prefixes of external ontologies that may be used
-    context = Context(data_model.get("prefixes") or {})
+    context = Context(project_definition.get("prefixes") or {})
 
-    # check if the project exists
-    project = None
+    # if project exists, update it, otherwise create it
+    project_local = Project(con=con, shortcode=project_definition["project"]["shortcode"])
     try:
-        project = Project(con=con, shortcode=data_model["project"]["shortcode"]).read()
+        project_remote: Project = try_network_action(
+            object=project_local,
+            method="read",
+            error_message_on_failure=""
+        )
```

Comment:
If the project can't be read, it raises a BaseError without message --> create the project.
If the reading suceeds without error --> update project